app/src/main/java/framgia/com/ichat/screen/privateroom/PrivateRoomFragment.java

```diff
@@ -62,9 +66,47 @@ public void onGetDataFailed() {
     }
 
     @Override
-    public void onGetListPrivateRoomSuccess(List<PrivateRoom> privateRooms) {
-        if (mPrivateRooms != null) {
-            mAdapter.addData(privateRooms);
+    public void onGetListPrivateRoomSuccess(List<Room> listRoom) {
+        if (mListRoom != null) {
```

Comment:
check null listRoom chứ nhỉ

Reply:
anh ơi.  cái  listRoom mà null thì 
 List<Room> listRoom = new ArrayList<>();
                for (DataSnapshot snapshot : dataSnapshot.getChildren()) {
                    Room room = snapshot.getValue(Room.class);
                    room.setId(snapshot.getKey());
                    listRoom.add(room);
                }
                mView.onGetListPrivateRoomSuccess(listRoom);
thì làm sao mà listRoom.add(room) đc đúng ko anh . vậy thì mình sẽ phải check như nào ạ

Reply:
Vậy đoạn này không cần check cũng được e nhé.
hoặc e có thể check trong presenter luôn trước khi chạy vòng for.
mặc dù không check thì không sao nhưng nó rỗng mà vẫn gọi đến các class khác và tốn thêm tài nguyên app e. 
code app phải tiết kiệm tài nguyên khổ lắm e à =))

Reply:
Em hiểu rồi anh ạ